src/components/DayPicker.jsx

```diff
@@ -424,18 +440,31 @@ class DayPicker extends React.Component {
   }
 
   onNextMonthClick(nextFocusedDate, e) {
-    const { isRTL } = this.props;
-    const { calendarMonthWidth } = this.state;
+    const { isRTL, numberOfMonths, daySize } = this.props;
+    const { calendarMonthWidth, currentMonth, monthTitleHeight } = this.state;
 
     if (e) e.preventDefault();
 
-    let translationValue = this.isVertical() ? -this.calendarMonthHeights[1] : -calendarMonthWidth;
+    const newMonth = currentMonth.clone().add(numberOfMonths, 'month');
+    const numberOfWeeks = getNumberOfCalendarMonthWeeks(newMonth, this.getFirstDayOfWeek());
+
+    let translationValue;
+
+    if (this.isVertical()) {
+      const calendarMonthWeeksHeight = this.calendarMonthWeeks[0] * (daySize - 1);
+      translationValue = -(monthTitleHeight + calendarMonthWeeksHeight + 1);
+    }
+
+    this.calendarMonthWeeks = this.calendarMonthWeeks.slice(1).concat([numberOfWeeks]);
```

Comment:
`const [, ...rest] = this.calendarMonthWeeks; this.calendarMonthWeeks = [...rest, numberOfWeeks];`?

Reply:
okay